source/testing/testing-components.md

```diff
@@ -25,74 +25,122 @@ export default Component.extend({
 
 ```tests/integration/components/pretty-color-test.js
-import { moduleForComponent, test } from 'ember-qunit';
+import { module } from 'qunit';
+import { setupRenderingTest } from 'ember-qunit';
+
+module('Integration | Component | pretty color', function(hooks) {
+  setupRenderingTest(hooks);
 
-moduleForComponent('pretty-color', 'Integration | Component | pretty color', {
-  integration: true
 });
 ```
+Inside of your `module` and after setting up the test, we can now start to create our first test case. Here we can use the `QUnit.test` helper and we can
+give it a descriptive name:
 
-Each test following the `moduleForComponent` call has access to the `render()`
-function, which lets us create a new instance of the component by declaring
-the component in template syntax, as we would in our application.
+```tests/integration/components/pretty-color-test.js
+import { module, test } from 'qunit';
+import { setupRenderingTest } from 'ember-qunit';
+
+module('Integration | Component | pretty color', function(hooks) {
+  setupRenderingTest(hooks);
 
-We can test that changing the component's `name` property updates the
-component's `style` attribute and is reflected in the  rendered HTML:
+  test('should change colors', async function(assert) {
+
+
+  });
+});
+```
+Also note how the callback function passed to the test helper is marked with the keyword `async`. The [ECMAScript 2017 feature async/await](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/await) allows us to write asynchronous code in an easy-to-read, seemingly synchronous manner. We can better see what this means, once we start writing out our first test case:
 
 ```tests/integration/components/pretty-color-test.js
-import { moduleForComponent, test } from 'ember-qunit';
+import { module, test } from 'qunit';
+import { setupRenderingTest } from 'ember-qunit';
+import { render } from '@ember/test-helpers';
 import hbs from 'htmlbars-inline-precompile';
 
-moduleForComponent('pretty-color', 'Integration | Component | pretty color', {
-  integration: true
+module('Integration | Component | pretty color', function(hooks) {
+  setupRenderingTest(hooks);
+
+  test('should change colors', async function(assert) {
+    assert.expect(1);
+
+    // set the outer context to red
+    this.set('colorValue', 'red');
+
+    await render(hbs`{{pretty-color name=colorValue}}`);
+
+    assert.equal(this.element.querySelector('div').getAttribute('style'), 'color: red', 'starts as red');
+  });
 });
 
-test('should change colors', function(assert) {
-  assert.expect(2);
+```
+
+Each test can use the `render()`
+function imported from the `@ember/test-helpers` package to create a new instance of the component by declaring
+the component in template syntax, as we would in our application.
```

Comment:
it does indeed "create a new instance", but it does that under the hood. what this thing does conceptually is "render" the component to the DOM.